timrlink.net.CLI/Actions/ProjectTimeDatabaseExportAction.cs

```diff
@@ -10,40 +13,99 @@
 
 namespace timrlink.net.CLI.Actions
 {
-    internal class ProjectTimeDatabaseExportAction
+    public class ProjectTimeDatabaseExportAction
     {
         private readonly DatabaseContext context;
         private readonly ILogger<ProjectTimeDatabaseExportAction> logger;
         private readonly IProjectTimeService projectTimeService;
         private readonly ITaskService taskService;
         private readonly IUserService userService;
+        private readonly string from;
+        private readonly string to;
+        private static string dateFormatToParse = "yyyy-MM-dd";
 
-        public ProjectTimeDatabaseExportAction(ILoggerFactory loggerFactory, string connectionString, IUserService userService, ITaskService taskService, IProjectTimeService projectTimeService)
+        public ProjectTimeDatabaseExportAction(ILoggerFactory loggerFactory, DatabaseContext context, string from, string to, IUserService userService, ITaskService taskService, IProjectTimeService projectTimeService)
         {
             logger = loggerFactory.CreateLogger<ProjectTimeDatabaseExportAction>();
-            context = new DatabaseContext(connectionString);
+            this.context = context;
+            this.from = from;
+            this.to = to;
             this.userService = userService;
             this.taskService = taskService;
             this.projectTimeService = projectTimeService;
         }
 
+        public static DateTime? TryParse(string text, string format) => 
+            DateTime.TryParseExact(text, format, CultureInfo.InvariantCulture, DateTimeStyles.None, out var date) ? date : (DateTime?) null;
+
         public async Task Execute()
         {
             logger.LogDebug("ProjectTimeDatabaseExportAction started");
-
-            await context.Database.EnsureCreatedAsync();
-
-            DateTime? lastProjectTimeImport = null;
-            var metadata = await context.GetMetadata(Metadata.KEY_LAST_PROJECTTIME_IMPORT);
-            if (DateTime.TryParseExact(metadata?.Value, "o", CultureInfo.InvariantCulture, DateTimeStyles.None, out var date))
+            
+            DateTime? fromDate = TryParse(from, dateFormatToParse);
+            DateTime? toDate = TryParse(to, dateFormatToParse);
+            
+            DateSpan? dateSpan;
+            
+            if (fromDate.HasValue && toDate.HasValue)
+            {
+                if (fromDate.Value > toDate.Value)
+                {
+                    throw new ArgumentException("From date is after to date. Aborting.");
+                }
+                
+                dateSpan = new DateSpan
```

Comment:
as mentioned earlier, why not add the to params to the constructor? (and make them immutable)